docs/docs/theme-ui.md

```diff
@@ -0,0 +1,264 @@
+
+## Using Theme UI in Gatsby
+
+Theme UI includes the `gatsby-plugin-theme-ui` package to better integrate with your Gatsby project.
+
+Install the following packages to add Theme UI.
+
+```shell
+npm install theme-ui gatsby-plugin-theme-ui @emotion/core @mdx-js/react
+```
+
+After installing the dependencies, add the following to your `gatsby-config.js`.
+
+```js:title=gatsby-config.js
+module.exports = {
+  plugins: ["gatsby-plugin-theme-ui"],
+}
+```
+
+Theme UI uses a `theme` configuration object to provide color, typography, layout, and other shared stylistic values through [React context][].
+This allows components within your site to add styles based on a predefined set of values.
+
+The Theme UI plugin uses the [component shadowing API][] to add the theme object context to your site.
+Create a `src/gatsby-plugin-theme-ui` directory in your project, and add an `index.js` file to export a theme object.
+
+```shell
+mkdir src/gatsby-plugin-theme-ui
+```
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {}
+```
+
+## Creating a theme object
+
+Add a `colors` object to the file created above to store the color palette for your site.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+}
+```
+
+Next add some base typographic values.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+  // highlight-start
+  fonts: {
+    body: "system-ui, sans-serif",
+    heading: "system-ui, sans-serif",
+    monospace: "Menlo, monospace",
+  },
+  fontWeights: {
+    body: 400,
+    heading: 700,
+    bold: 700,
+  },
+  lineHeights: {
+    body: 1.5,
+    heading: 1.125,
+  },
+  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 72],
+  // highlight-end
+}
+```
+
+Next add values for use in margin and padding.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+  fonts: {
+    body: "system-ui, sans-serif",
+    heading: "system-ui, sans-serif",
+    monospace: "Menlo, monospace",
+  },
+  fontWeights: {
+    body: 400,
+    heading: 700,
+    bold: 700,
+  },
+  lineHeights: {
+    body: 1.5,
+    heading: 1.125,
+  },
+  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 72],
+  // highlight-start
+  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
+  // highlight-end
+}
+```
+
+Feel free to include as many additional values as you'd like in your theme.
+Read more about theming in the [Theme UI documentation](https://theme-ui.com/theming).
+
+## Adding styles to elements
+
+Theme UI uses a custom [JSX pragma][] comment to add support for Emotion's `css` prop,
+which can be used to style elements.
+By default JSX compiles to the `React.createElement` function.
```

Comment:
I feel like these two sentences about JSX would be great as an advanced callout. @amberleyromo your idea about little graphic icons in the docs would fit well here. But maybe there's something we can add in the meantime, like a note? My concern is that people will come to this section and think "this is too advanced for me" and give up early.

Reply:
Yeah, this part of JSX is not the easiest part to explain. I added more in this section based on @amberleyromo's suggestion, but I'm not sure how detailed to go here. I think a special callout would make sense – do we have a way to do that today?

Reply:
Perhaps showing code might be the most effective. If it's a necessary concept, can you try minimal code blocks to show the difference between the `React.createElement` and pragma output? Similar to line 128 with `jsx('button')`, but called out a bit more. If line 129 about Theme UI's `jsx` function could be shown with code, that would help too.

Reply:
I'm taking another look at this now, and I feel like going into how JSX works is maybe a little too advanced for the concepts this doc is trying to introduce. I think with the first line reworked a little, we could avoid explaining JSX and perhaps we could link to https://jasonformat.com/wtf-is-jsx/ instead of going into too much detail here

Reply:
So I love the article you've linked here, but I think it _really_ doesn't solve the problem of explaining this on a more approachable level to beginner folks. We would need to explicitly mark that as an advanced read.